### PR TITLE
Add Complex.sign to v6.e

### PR DESCRIPTION
**Pull request by coke**
Inspired by https://www.reddit.com/r/rakulang/comments/10sm0sm/why_isnt_sign_defined_for_complex_numbers/

### CODE REVIEW

**Review by lizmat — APPROVED** [review]

**Comment by vrurg** [comment]
Any reason not to have it in 6.c? It doesn't break backward compatibility.

**Comment by 2colours** [comment]
Don't take this at an instant objection but as a general rule, please don't forget that "breaking backwards compatibility" is essentially subjective. Anything that isn't specified to behave in a certain way may be depended upon at any moment - so even though I agree it would be weird to depend on `$number.sign` breakage for `Complex` values, a stricter sense of "compatibility" might demand that behavior like this might not change with a compiler release.

**Comment by lizmat** [comment]
I think we also need people to incentivize to use v6.e  :-)

**Comment by p6steve** [comment]
this commit will fail

```
ok (0 + 0i).sign == 0,      'sign of 0+0i is 0';
```

File: src/core.e/Complex.pm6 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+augment class Complex {
+    method sign() { self / self.abs }
```
Comment (markjreed) [inline src/core.e/Complex.pm6:2]: You need a check for 0 here. sign(0+0i) should be 0, not trigger a division-by-zero error.

**Comment by coke** [comment]
Fixed sign(0i) and added a signature similar to Int's version.

**Comment by coke** [comment]
> ```
> ok (0 + 0i).sign == 0,      'sign of 0+0i is 0';
> ```

Note that the test given above works even in v6.c, since the imaginary part is 0, it gets coerced to something Cool that works. :)

**Comment by lizmat** [comment]
Thanks!

**Pull request merged**